screeps-game-api/src/objects/impls/room_terrain.rs

```diff
@@ -1,15 +1,37 @@
-use {constants::Terrain, objects::RoomTerrain};
+use stdweb::UnsafeTypedArray;
+
+use {constants::{ReturnCode, Terrain}, objects::RoomTerrain, traits::TryInto};
 
 impl RoomTerrain {
     pub fn constructor(room_name: &str) -> Self {
-        js_unwrap!{new Room.Terrain(@{room_name})}
+        js_unwrap!(new Room.Terrain(@{room_name}))
     }
 
     pub fn get(&self, x: u32, y: u32) -> Terrain {
-        js_unwrap!{@{self.as_ref()}.get(@{x}, @{y})}
+        js_unwrap!(@{self.as_ref()}.get(@{x}, @{y}))
     }
 
     pub fn get_raw_buffer(&self) -> Vec<u8> {
-        js_unwrap!(@{self.as_ref()}.getRawBuffer())
+        let mut buffer: Vec<u8> = vec![0; 2500];
+        self.get_raw_buffer_array_ref(&mut buffer).expect("Panic in get_raw_buffer.");
+        buffer
     }
+
+    pub fn get_raw_buffer_array_ref<'a>(&self, buffer: &'a mut Vec<u8>) -> Result<&'a mut Vec<u8>, ReturnCode> {
+        let is_success: bool;
+        {
+            let arr: UnsafeTypedArray<u8> = unsafe{ UnsafeTypedArray::new(buffer.as_mut_slice()) };
+
+            is_success = js!{
+                var bytes = @{arr};
+                return @{self.as_ref()}.getRawBuffer(bytes) != -10;
```

Comment:
Could we possibly unwrap a `ReturnCode`, or at least compare to `0` rather than `-10`? Documentation states we'll only get `-10` but that could change in the future and the server doesn't always follow the spec. I think it would be better to err on the side of reporting failure rather than seeing any non-`-10` error code as a success.

Reply:
I can't compare with zero as success is defined as returning the array. I could potentially compare it with `bytes` to make sure it's the same ref? I'll have to look this up though.

Reply:
A quick look seems to indicate that it'd compare all 2500 values :/